### DISCUSSION TITLE
Feature request: Text selection in Log and RichLog

### DISCUSSION THREAD

**Discussion by mensinda** [Textualize/textual#5334] [Ideas]
Hi,

thank you for developing textual!

It would be nice if `Log` and `RichLog` had a built-in text selection feature, similar to `TextArea`. This would make copying logs a lot more user friendly.

---

I know that it is possible select text [directly with the terminal](https://textual.textualize.io/FAQ/#how-can-i-select-and-copy-text-in-a-textual-app), however, this approach has some drawbacks:

- Scrolling the log does not automatically scroll the selection
- A multiline selection goes over the entire with of the screen and not the widget content

**Comment by willmcgugan** [comment]
We have a more comprehensive selection feature planned. It should cover just about everything, much like a web page.

**Reply by stavalfi** [reply]
hi @willmcgugan any update on that? 🗡️

**Reply by TomJGooding** [reply]
I'm not Will, but Textual now supports text selection for most widgets.

https://textual.textualize.io/FAQ/#how-can-i-select-and-copy-text-in-a-textual-app

**Reply by stavalfi** [reply]
Hi this is not the same as default selection 😬

**Reply by TomJGooding** [reply]
What do you mean by "default selection"? The original discussion was about selecting text, so it really isn't clear what you're asking.
